segmentstore/server/src/test/java/io/pravega/segmentstore/server/writer/TestWriterDataSource.java

```diff
@@ -273,6 +274,16 @@ public InputStream getAppendData(long streamSegmentId, long startOffset, int len
         }
 
         synchronized (this.lock) {
+            // Perform the same validation checks as the ReadIndex would do.
```

Comment:
Just so that I understand: in the case we change these checks in the `ReadIndex`, do we have to remember to do it here too? Could we avoid the duplication?

Reply:
This `TestWriterDataSource` is an emulation of the entire world outside of the `StorageWriter` (`ReadIndex` + `DurableLog`). I don't see an easy way to share 3 lines of code between such remote components.